### PR TITLE
Fix getAmountIn round-trip test to use realistic reserves (closes #83)

### LINKED ISSUE #83: dex/v2 getAmountIn inverse: round-trip assumption fails for small reserve_out
Discovered via the test-harness fix. getAmountIn(getAmountOut(x))>=x within 2 units fails for reserve_out=200e9 vs reserve_in=100e18. getAmountIn already applies the Uniswap +1 ceiling, so this is likely the test's within-2-units assumption breaking in the small-output regime, not an impl bug. Needs analysis. Quarantined with error.SkipZigTest pointing here.

### PR DESCRIPTION
**Pull request by koko1123**
## What
The `getAmountIn`/`getAmountOut` implementations are correct standard Uniswap V2. The quarantined `getAmountIn inverse` test used `reserve_out = 200e9` against `reserve_in = 100e18` -- a 1e9:1e20 imbalance.

In that regime `getAmountOut` floors a small output (1974316068.x -> 1974316068), discarding ~4e8 wei of input-equivalent. `getAmountIn` then computes the input for *exactly* that floored output, which is ~4e8 below the original `amount_in` -- the +1-unit ceiling can't bridge that gap. That's correct AMM behaviour; the test's "recovered >= amount_in within 2 units" assumption only holds when the output is large.

## Fix
Use balanced reserves (`reserve_out = 200e18`); the floored output is then large and the round-trip is exact (diff 0). Documented the invariant's precondition. No production code changed -- #83 was a bad test.

## Result
**The unit suite now runs with zero skips: 775/775 pass.** All four bugs the test-harness fix (#84) surfaced are resolved (#80 keccak fallback and #81 mulDiv were real code bugs; #82 and #83 were bad tests). 0.16.0 and 0.17-dev green.

Closes #83

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected DEX input/output calculation and inverse round-trip validation
  * Fixed BIP-39 mnemonic seed / Trezor passphrase test vectors
  * Resolved mulDiv correctness at large integer boundary cases
  * Fixed pure-Zig Keccak-256 fallback behavior

* **Tests**
  * Test suite now runs fully with no skipped tests (775/775); previously quarantined issues have been resolved

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [Unreleased]
 
 ### Fixed
+- DEX `getAmountIn`/`getAmountOut` are correct standard Uniswap V2; the `getAmountIn inverse` round-trip test used unrealistic tiny-reserve parameters where flooring a small output legitimately breaks the within-2-units invariant. Corrected the test to balanced reserves and un-quarantined it (#83). The full unit suite now runs with zero skips.
```
[the remark this answers is not in the document]
Comment (koko1123) [reply CHANGELOG.md:11]: Good catch - fixed. The stale clause now reads that all four are resolved (2 real code bugs, 2 bad tests) with zero skips, consistent with the per-issue Fixed entries.
